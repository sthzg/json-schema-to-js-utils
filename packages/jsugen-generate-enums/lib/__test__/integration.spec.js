import schema from '@sthzg/jsugen-core/lib/sources/jsonSchema/testResources/example.schema';
import contactSchema from '@sthzg/jsugen-core/lib/sources/jsonSchema/testResources/example.contact';
import { generateEnumsModule as generate } from '../index';

describe('generate-enums', () => {
  test('should pass full integration test', done => {
    expect.assertions(1);

    generate({ schema }).subscribe(output => {
      expect(output).toMatchSnapshot();
      done();
    });
  });

  test('should pass example.contact.json integration test', done => {
    expect.assertions(1);

    generate({ schema: contactSchema }).subscribe(output => {
      expect(output).toMatchSnapshot();
      done();
    });
  });
});
