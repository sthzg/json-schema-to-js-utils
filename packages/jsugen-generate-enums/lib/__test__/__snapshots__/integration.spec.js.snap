// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate-enums integration test 1`] = `
"/**
 * @file This file is auto-generated. Do not modify its contents manually.
 */

export const RootLevelEnumArrayValues = ['FOO', 'BAR'];
export const RootLevelEnumArrayEnum = {
  FOO: 'FOO',
  BAR: 'BAR',
};

export const LevelOneAnyEnumArrayValues = ['ONE.BAM', 'ONE.BAZ'];
export const LevelOneAnyEnumArrayEnum = {
  'ONE.BAM': 'ONE.BAM',
  'ONE.BAZ': 'ONE.BAZ',
};
"
`;
